### DISCUSSION TITLE
Block Unwanted Participants

### DISCUSSION THREAD

**Discussion by nexquery** [jchristn/SuperSimpleTcp#196] [Q&A]
Hello, people I do not know can connect to my TCP server without using a client. To prevent this, I want it to perform a verification before connecting to the server, but I have no idea how to do this. I think I need to use a certificate like simpletcp.crt. If this is done with a certificate, can you give me a short example of how to do it?

**Comment by jchristn** [comment]
Hi @nexquery you could use SSL certificates for this, you would have to install them in the server's operating system for each client.  Another approach would be to use ```Settings.PermittedIPs``` to specify the IP addresses you would like to allow to connect (if known a priori).  Yet another approach would be, within whatever messaging architecture you build on top of this library, to authenticate requestors prior to allowing them to interact.
